test/parallel/test-whatwg-url-properties.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+require('../common');
+
+const URL = require('url').URL;
+const assert = require('assert');
+
+const url = new URL('http://user:pass@foo.bar.com:21/aaa/zzz?l=24#test');
+const oldParams = url.searchParams;  // for test of [SameObject]
+
+// To retrieve enumerable but not necessarily own properties,
+// we need to use the for-in loop.
+const props = [];
+for (const prop in url) {
+  props.push(prop);
+}
+
+// See: https://url.spec.whatwg.org/#api
+// https://heycam.github.io/webidl/#es-attributes
+// https://heycam.github.io/webidl/#es-stringifier
+const expected = ['toString',
+  'href', 'origin', 'protocol',
+  'username', 'password', 'host', 'hostname', 'port',
+  'pathname', 'search', 'searchParams', 'hash'];
+
+assert.deepStrictEqual(props, expected);
```

Comment:
I wonder: Is there a chance the ordering of the properties changes? Would we want to catch that or do we want to `.sort()` these?